sync/tc.py

```diff
@@ -411,6 +413,25 @@ def lookup_index(index_name):
     return task_id
 
 
+def lookup_treeherder(project, revision):
+    push_data = fetch_json(TREEHERDER_BASE + "api/project/%s/push/" % (project,),
+                           params={"revision": revision})
+
+    pushes = push_data.get("results", [])
+    push_id = pushes[0].get("id") if pushes else None
+    if push_id is None:
+        return
+
+    jobs_data = fetch_json(TREEHERDER_BASE + "api/jobs/",
+                           {"push_id": push_id})
+    property_names = jobs_data["job_property_names"]
+    idx_name = property_names.index("job_type_name")
+    idx_task = property_names.index("task_id")
+    decision_tasks = [item for item in jobs_data.get("results", [])
+                      if item[idx_name] == "Gecko Decision Task"]
+    return decision_tasks[-1][idx_task]
```

Comment:
Can jobs_data ever not have the `results` key? If so then this could fail since `decision_tasks` would be an empty list.

Reply:
I think "no" is the real answer here. It would be much nicer is we validated the input close to the edges and then could assume correct input in the rest of the code (something, something, proper type system).